### DISCUSSION TITLE
Source Code?

### DISCUSSION THREAD

**Discussion by Rusty-Weasel** [rustdesk/rustdesk-server-pro#42] [Q&A] [answered]
Please explain this situation:

**rustdesk-server-pro** is a clone plus a **bit more features** from **rustdesk-server**,

and **rustdesk-server** is licensed under **APGL-3.0** so the **rustdesk-server-pro** also needs to be open source in my mind, 
as the sources of rustdesk-server **are distributed by rustdesk-server-pro**

![image](https://github.com/rustdesk/rustdesk-server-pro/assets/61587283/3fbecfc5-032b-466d-89be-1856659c4468)
https://choosealicense.com/licenses/agpl-3.0/

![image](https://github.com/rustdesk/rustdesk-server-pro/assets/61587283/39abd8cc-2385-4b1e-8b38-81e5a23e8b40)
https://fossa.com/blog/open-source-software-licenses-101-agpl-license/

![image](https://github.com/rustdesk/rustdesk-server-pro/assets/61587283/85b80054-6f6b-456a-84f4-8a7614979548)
https://snyk.io/learn/agpl-license/

**So if a customer buy a license it is distributed outside of your organisation**,
and the customer and/or his users can **access server-side application** which is normally public facing,
so customer can expect source code, or did i miss something?

Awaiting your explanation?

**Comment by Rusty-Weasel** [comment]
@rustdesk not willingness to give us a feedback ? why you move issue to discussion without comment?

**Reply by DrRuhe** [reply]
Obligatory I am not a Lawyer and not affilliated with rustdesk. Furthermore, I am assuming that none of their dependencies are licensed to them under AGPL terms (I have not explicitly checked) and that contributions to the project get transferred in ownership to the rustdesk project.

Since they are the owners of the rustdesk-server program they can do whatever they want with it. In principle the owner of a piece of software can issue licenses however they please. This includes mutiple licenses (all the OSS repos licensed under MIT/Apache 2.0 are an example). So they could simply issue a more permissive license for themselves.

**Comment by Rusty-Weasel** [comment]
![image](https://github.com/rustdesk/rustdesk-server-pro/assets/61587283/3cac2a82-0c36-457c-9386-9ad2b59264eb)
see old screenshot, 
pro server where changed to none some weeks ago

here the initial commit of pro server:
![image](https://github.com/rustdesk/rustdesk-server-pro/assets/61587283/11b8595d-1e96-44a8-81fa-c19915ddb73f)

and after i mentioned it, as it was clear that it will be closed source they changed it:
![image](https://github.com/rustdesk/rustdesk-server-pro/assets/61587283/a55b4e85-6ded-4d9b-bb98-d73a505d979c)

**Comment by Rusty-Weasel** [comment]
AGPL:
**Modify the code:** 
Users can change or rework the code, but if they distribute these changes/modifications in any public way (including over a server), they must release these updates in source code form under the AGPL license.

**Distribute copies or modified/updated versions of the code**: 
As long as these modifications are also released under the GNU AGPL, they can be distributed to others.

**Both parts have been done with the pro server edition, it is a modified and extended version!**

and till today there was no explanation concerning this.

**Comment by otonoton** [comment]
I have opened this issue on the matter: https://github.com/rustdesk/rustdesk-server-pro/issues/781

**Reply by rustdesk** [reply]
You are blocked, since you posted duplicate topics.
https://github.com/rustdesk/rustdesk-server/issues/596
and
https://github.com/rustdesk/rustdesk-server-pro/issues/781

**Comment by rustdesk** [comment, marked as answer]
Explained here: https://github.com/rustdesk/rustdesk/discussions/5942#discussioncomment-14523104
